cmd/dockerd/daemon.go

```diff
@@ -227,13 +228,18 @@ func (cli *DaemonCli) start(opts daemonOptions) (err error) {
 
 	name, _ := os.Hostname()
 
+	// Use a buffered channel to pass changes from store watch API to daemon
+	// A buffer allows store watch API and daemon processing to not wait for each other
+	watchStream := make(chan *swarmapi.WatchMessage, 32)
```

Comment:
It was decided to add a `watch` API to expose cluster change. But do not add a separate `LogEvent` interface for simplicity. 
https://github.com/docker/swarmkit/issues/491#issuecomment-282173645